src/lwip/app_ethernet.c

```diff
@@ -54,11 +61,69 @@
 uint32_t EthernetLinkTimer;
 
 #if LWIP_DHCP
-#define MAX_DHCP_TRIES 4
+#define MAX_DHCP_TRIES 20
 uint32_t DHCPfineTimer = 0;
 uint8_t DHCP_state = DHCP_OFF;
 #endif
 
+void print_netif_ip(struct netif *netif) {
+  // Ensure that the netif is valid and has an IP address assigned
+  if (netif != NULL && netif_is_up(netif) && netif->ip_addr.addr != 0) {
+    // Convert the IP address to a human-readable string format
+    char ip_str[16];
+    ip4addr_ntoa_r(&netif->ip_addr, ip_str, sizeof(ip_str));
+    // Print the IP address
+    printf("IP Address: %s\n", ip_str);
+  } else {
+    printf("No IP Address assigned to the netif.\n");
+  }
+}
```

Comment:
Can we safely remove this? I don't see any added value.

Reply:
Could be useful if you want to read it through the debugger for the ip address that was obtained. Let me know if you want it gone I can remove it

Reply:
Yea it can be useful in the future when we develop out our system configuration structs. But in the meantime we wont use it and it wouldn't be hard to reimplement this when we need it.